single_cell/qc.py

```diff
@@ -0,0 +1,222 @@
+        ctx={'docker_image': config['docker']['single_cell_pipeline']}
+    )
+
+    workflow.setobj(
+        obj=mgd.OutputChunks('sample_id', 'library_id'),
+        value=list(samplegroups),
+    )
+
+
+    workflow.subworkflow(
+        name='create_sample_level_plots',
+        func="single_cell.workflows.qc.create_sample_level_plots",
+        axes=('sample_id', 'library_id', ),
+        args=(
+            mgd.InputFile('sample_ids', 'sample_id', 'library_id', fnames=sample_ids),
+            mgd.InputFile('library_ids', 'sample_id', 'library_id', fnames=library_ids),     
+            mgd.InputFile('mappability', 'sample_id', 'library_id', fnames=mappability_files),
+            mgd.InputFile('strelka', 'sample_id', 'library_id', fnames=strelka_files),
+            mgd.InputFile('museq', 'sample_id', 'library_id', fnames=museq_files),
+            mgd.InputFile('cosmic_status', 'sample_id', 'library_id', fnames=cosmic_status_files),
+            mgd.InputFile('snpeff', 'sample_id', 'library_id', fnames=snpeff_files),
+            mgd.InputFile('dbsnp_status', 'sample_id', 'library_id', fnames=dbsnp_status_files),
+            mgd.InputFile('trinuc', 'sample_id', 'library_id', fnames=trinuc_files),
+            mgd.InputFile('counts', 'sample_id', 'library_id', fnames=counts_files),
+            mgd.InputFile('breakpoint_annotation', 'sample_id', 'library_id', fnames=breakpoint_annotation),
+            mgd.InputFile('breakpoint_counts', 'sample_id', 'library_id', fnames=breakpoint_counts),
+            mgd.InputFile('haplotype_allele_data', 'sample_id', 'library_id', fnames=haplotype_allele_data),
+            mgd.InputFile('annotation_metrics', 'sample_id', 'library_id', fnames=annotation_metrics),
+            mgd.InputFile('hmmcopy_reads', 'sample_id', 'library_id', fnames=hmmcopy_reads),
+            mgd.InputFile('hmmcopy_segs', 'sample_id', 'library_id', fnames=hmmcopy_segs),
+            mgd.InputFile('hmmcopy_metrics', 'sample_id', 'library_id', fnames=hmmcopy_metrics),
+            mgd.InputFile('alignment_metrics', 'sample_id', 'library_id', fnames=alignment_metrics),
+            mgd.InputFile('gc_metrics', 'sample_id', 'library_id', fnames=gc_metrics),       
+            mgd.InputFile('indel_files', 'sample_id', 'library_id', fnames=indel_files),       
+            mgd.OutputFile('sample_level_report_htmls', 'sample_id', 'library_id', 
+                fnames=sample_level_report_htmls
+            ),
+            mgd.OutputFile('mafs', 'sample_id', 'library_id', fnames=mafs),
+            mgd.OutputFile('snvs_all', 'sample_id', 'library_id', fnames=snvs_all),
+            out_dir,
+            mgd.InputFile('outpaths', 'sample_id', 'library_id', fnames=outpaths),
+            vep_refdir
+        ),
+    )
+    
+    workflow.subworkflow(
+        name='create_patient_workflow',
+        func="single_cell.workflows.qc.create_patient_workflow",
+        args=(
+            patient,
```

Comment:
is this a file??

Reply:
no

Reply:
see line 57. i think this is a file.

Reply:
it isnt a file, its a dictionary of strings - i just didnt know how else to do it